### PR TITLE
(MODULES-2062) updates prefork.conf params for apache 2.4

### PR DESCRIPTION
**Pull request by eputnam**
MaxClients is now MaxRequestWorkers and MaxRequestsPerChild is now MaxConnectionsPerChild. This updates the manifest, template, and adds a couple unit tests. Unit tests for every OS did not seem necessary since all that is being tested is an if block.

### CODE REVIEW

**Comment by tphoney** [comment]
Cheers !

File: manifests/mod/prefork.pp [hunk 1]
```diff
@@ -23,6 +25,15 @@
   if defined(Class['apache::mod::worker']) {
     fail('May not include both apache::mod::prefork and apache::mod::worker on the same node')
   }
+
+  if versioncmp($_apache_version, '2.3.13') < 0 {
```
Comment (ekohl) [inline manifests/mod/prefork.pp]: Shouldn't this be ```>= 0```?

File: templates/mod/prefork.conf.erb [hunk 2]
```diff
@@ -3,6 +3,14 @@
   MinSpareServers     <%= @minspareservers %>
   MaxSpareServers     <%= @maxspareservers %>
   ServerLimit         <%= @serverlimit %>
-  MaxClients          <%= @maxclients %>
-  MaxRequestsPerChild <%= @maxrequestsperchild %>
+  <%- if @maxrequestworkers -%>
+  MaxRequestWorkers      <%= @maxrequestworkers %>
+  <%- elsif @maxclients -%>
+  MaxClients             <%= @maxclients %>
```
Comment (ekohl) [inline templates/mod/prefork.conf.erb]: Would you mind keeping the whitespace? That results in a smaller diff on clients.

File: templates/mod/prefork.conf.erb [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
+  <%- end -%>
+  <%- if @maxconnectionsperchild -%>
+  MaxConnectionsPerChild <%= @maxconnectionsperchild %>
+  <%- elsif @maxrequestsperchild -%>
+  MaxRequestsPerChild    <%= @maxrequestsperchild %>
```
Comment (ekohl) [inline templates/mod/prefork.conf.erb]: Same here.

**Pull request merged**